vmm/src/lib.rs

```diff
@@ -1674,6 +1674,7 @@ impl Vmm {
                     kvm_hyperv: vm_config.lock().unwrap().cpus.kvm_hyperv,
                     #[cfg(feature = "tdx")]
                     tdx_enabled: vm_config.lock().unwrap().is_tdx_enabled(),
+                    amx_enabled: vm_config.lock().unwrap().cpus.features.amx,
```

Comment:
This may cause deadlock 'cause vm_config.lock() have not been unlocked in kvm_hyperv.

Reply:
You are right! Also, it unveiled an existing race condition when tdx is enabled. It was never noticed, as we haven't implemented the snapshot/migration for TDX. Sent another PR for fixing it: #5845. PTAL. Thank you!.